src/averageAges.js

```diff
@@ -37,7 +44,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const reducer = (arr) => arr.reduce((elem, total) => elem + total, 0);
+
+  const mother = people
+    .filter((e) => e.sex === 'f' && people.find(s => s.mother === e.name))
+    .map((e) => e.died - e.born);
+  const femail = people
+    .filter((e) => e.sex === 'f')
+    .map((e) => e.died - e.born);
+
+  return withChildren
+    ? reducer(mother) / mother.length
+    : reducer(femail) / femail.length;
```

Comment:
do it like in the first function where you first determine the array you want to use based on `withChildren` and then you run the `reducer` only once and return it